src/EnergyPlus/Coils/CoilCoolingDXCurveFitSpeed.cc

```diff
@@ -0,0 +1,378 @@
+#include <Coils/CoilCoolingDXCurveFitOperatingMode.hh>
+#include <Coils/CoilCoolingDXCurveFitSpeed.hh>
+#include <Coils/PsychStruct.hh>
+#include <CurveManager.hh>
+#include <DataEnvironment.hh>
+#include <DataHVACGlobals.hh>
+#include <DataIPShortCuts.hh>
+#include <DataPrecisionGlobals.hh>
+#include <DataSizing.hh>
+#include <DXCoils.hh>
+#include <InputProcessor.hh>
+#include <Psychrometrics.hh>
+#include <ReportSizingManager.hh>
+
+using namespace EnergyPlus;
+using namespace DataIPShortCuts;
+
+void CoilCoolingDXCurveFitSpeed::instantiateFromInputSpec( CoilCoolingDXCurveFitSpeedInputSpecification input_data, CoilCoolingDXCurveFitOperatingMode * _parentMode ) {
+    this->original_input_specs = input_data;
+    this->parentMode = _parentMode;
```

Comment:
@Myoldmopar Running with new test file HVACTemplate-5ZoneUnitarySystem-SingleSpeed-exp.idf which has 5 unitary systems and 5 dx coils. At this point, coil1 speed 1 "Sys 1 Furnace DX Cool Cooling Coil Speed 1 Performance" thinks its parent is the correct one "Sys 1 Furnace DX Cool Cooling Coil Operating Mode".